hy/models.py

```diff
@@ -56,13 +56,22 @@ class HyObject(object):
     def replace(self, other, recursive=False):
         if isinstance(other, HyObject):
             for attr in self.properties:
-                if not hasattr(self, attr) and hasattr(other, attr):
+                # Can't use hasattr() to check if attributes are set because
+                # hasattr() will call our __getattr__() implementation.
+                # Thus we check self.__dict__ explicitly.
+                if attr not in self.__dict__ and attr in other.__dict__:
                     setattr(self, attr, getattr(other, attr))
         else:
             raise TypeError("Can't replace a non Hy object '{}' with a Hy object '{}'".format(repr(other), repr(self)))
 
         return self
 
+    def __getattr__(self, attr):
+        if attr in ("start_line", "start_column"):
+            return 1
+        else:
+            return super().__getattr__(attr)
+
```

Comment:
Couldn't a `property` be used for these two?  That would be less intense than overriding `__getattr__`.

Reply:
That's entirely reasonable, and I'm somewhat embarrassed for not having thought about that.
Updated and rebased, thanks for the catch!